torch/csrc/jit/pickler.cpp

```diff
@@ -1017,15 +1031,27 @@ PickleOpCode Unpickler::readInstruction() {
       if (device_) {
         device = *device_;
       }
-      at::DataPtr storage_ptr = read_record_(key);
       int64_t numel = args.at(4).toInt();
-      at::Storage storage(
-          at::CPU(type).typeMeta(),
-          numel,
-          std::move(storage_ptr),
-          /*allocator=*/nullptr,
-          /*resizable=*/false); // NB: we didn't set any allocator for the
-                                // tensor
+
+      bool read_tensors_later = read_record_ == nullptr;
```

Comment:
It doesn't make sense to add this as an option. Once one use does it this way, it is simpler for all uses to do it this way.